pyccel/ast/core.py

```diff
@@ -3149,46 +3149,58 @@ def __init__(
 
     @property
     def name(self):
+        """ Name of the function """
         return self._name
 
     @property
     def arguments(self):
+        """ List of variables which are the function arguments """
         return self._arguments
 
     @property
     def results(self):
+        """ List of variables which are the function results """
         return self._results
 
     @property
     def body(self):
+        """ CodeBlock containing all the statements in the function """
         return self._body
 
     @property
     def local_vars(self):
+        """ List of variables defined in the function """
         return self._local_vars
 
     @property
     def global_vars(self):
+        """ List of global variables used in the function """
         return self._global_vars
 
     @property
     def cls_name(self):
+        """ String containing the name of the class to which the method belongs.
+        If the function is not a class procedure then this returns None """
         return self._cls_name
 
     @property
     def hide(self):
+        """ False if function is pyccelized, True otherwise """
         return self._hide
 
     @property
     def kind(self):
+        """ Returns function or procedure """
         return self._kind
```

Comment:
@saidctb Is this property useful? It returns either 'procedure' or 'function'. So I am not sure what it does. In fortran a 'procedure' is either a `function` or a `subroutine`. In pyccel it seems that `kind` is always 'function'. It is used in the function `is_procedure`. As far as I can tell `is_procedure` returns True if the FunctionDef is a `subroutine` (in any case this is how it is used in fcode)

Reply:
it's not useful